be/src/exec/parquet/file_reader.cpp

```diff
@@ -58,23 +59,41 @@ Status FileReader::get_next(vectorized::ChunkPtr* chunk) {
 }
 
 Status FileReader::_parse_footer() {
-    // try
+    // try with buffer on stack
     constexpr uint64_t footer_buf_size = 16 * 1024;
+    uint8_t local_buf[footer_buf_size];
```

Comment:
Maybe you can use `boost/container/small_vector.hpp` or something alike.

Reply:
Thanks, but I think it will things complicated. 

We just need a temporary buffer to read out the magic number and footer metadata right here, then discard it in the same function. No extra operations like resize or iterator something.